mParticle-OneTrust/MPKitOneTrust.m

```diff
@@ -55,6 +73,64 @@ - (void)start {
     });
 }
 
+// Listener for Consent Change dispatched by One Trust
+-(void)actionConsent:(NSNotification*)notification {
+    NSString *category = notification.name; // Cookie Name
+    NSNumber *status = notification.object; // BOOL-ish value for consent
+    category = notification.name;
+
+    // Fire consent change event
+    [self createConsentEvent:category :status];
+}
+
+// Parses the raw consent mapping from the mParticle UI into simple map
+- (NSDictionary*)parseConsentMapping:(NSString*)rawConsentMapping {
+    NSData *jsonData = [rawConsentMapping dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error;
+
+    NSMutableDictionary *consentMapping = [[NSMutableDictionary alloc] init];
+
+    id jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
+
+        if (error) {
+            NSLog(@"Error parsing JSON: %@", error);
+        }
+        else
+        {
+            if ([jsonObject isKindOfClass:[NSArray class]])
+            {
+                for (NSDictionary *element in jsonObject) {
+                    consentMapping[element[@"value"]] = element[@"map"];
+                }
+            }
+            // TODO: Should we throw an error if the consent mapping
+            //       is not an array?
+            // else {
+            //     NSDictionary *jsonDictionary = (NSDictionary *)jsonObject;
+            //     NSLog(@"jsonDictionary - %@",jsonDictionary);
+            // }
+        }
+    return consentMapping;
+}
+
+// Creates an mParticle Consent Event
+-(void)createConsentEvent:(NSString*)cookieName
+                         :(NSNumber*)status {
+    // TODO: Can we move this out of the function?
```

Comment:
I'd be fine with it being part of actionConsent: above

Reply:
I made it a separate method because we need to call it inside the actionConsent and on init.

Reply:
ahh yea I see that now. Then yeah I think everything being done in this function deserves to be here.